File: scala/src/main/scala/folio/codinginterview/application/repository/MarketPriceRepository.scala
package folio.codinginterview.application.repository

import folio.codinginterview.domain.StockSymbol
import scala.concurrent.Future

/** 市場価格リポジトリ。 */
trait MarketPriceRepository {
  def all(): Future[Map[StockSymbol, BigDecimal]]
  def update(prices: Map[StockSymbol, BigDecimal]): Future[Unit]
}

File: java8/src/main/java/folio/codinginterview/application/repository/MarketPriceRepository.java
package folio.codinginterview.application.repository;

import folio.codinginterview.domain.StockSymbol;

import java.math.BigDecimal;
import java.util.Map;
import java.util.concurrent.CompletableFuture;

/**
 * 市場価格リポジトリ。
 */
public interface MarketPriceRepository {
    CompletableFuture<Map<StockSymbol, BigDecimal>> all();

    CompletableFuture<Void> update(Map<StockSymbol, BigDecimal> prices);
}
